### ISSUE TITLE
A fixed tensor is divided into two parts for two GPU while in multi-gpu env for training with device=[0, 1].

### ISSUE THREAD

**Issue by Dandelionym** [pytorch/pytorch#82932]
### 🐛 Describe the bug

Hi, here is a problem.

The initialization process is simply given as below and the adjs is a numpy array:
`supports=(torch.tensor(adjs)).type(torch.float32).to(device)`
Then, I run code in 2 gpu:
`net = nn.DataParallel(net)`

After that, in the forward process, a problem occurred in this line:
`A=self.h+supports`
where self.h = self.h=Parameter(torch.zeros(N,N), requires_grad=True) and N is a fixed number in the settings.

The error shows that, variable `supports` is divided into two parts at dim 1, which leads to the end of dimension mismatch between self.h and supports. So the operation of add is failed here.

Please tell me the truth and what I should do. Thanks a lot! :-)

### Versions

PyTorch 1.8.1-cu11

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @kwen2501

**Comment by DongyuXu77** [comment]
Hi, I'm a newbie to this community and sincerely hope to take some work on it. Can I get a chance to handle this bug, what should I do?

**Comment by ngimel** [comment]
DataParallel is deprecated, so we are not looking for any changes to it #65936

**Comment by rohan-varma** [comment]
As @ngimel mentioned, we are recommending use of `DistributedDataParallel` over `DataParallel`. Could you please try `DistributedDataParallel` for your workload and let us know if you face any issues?
